sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -423,3 +425,91 @@ async fn test_leased_blob_operations(ctx: TestContext) -> Result<(), Box<dyn Err
+    container_client.create_container(None).await?;
+    create_test_blob(&blob_client).await?;
+
+    // Set Legal Hold
+    blob_client.set_legal_hold(true, None).await?;
+    let response = blob_client.get_properties(None).await?;
+    // Assert
+    let legal_hold = response.legal_hold()?;
+    assert!(legal_hold.unwrap());
+
+    // Attempt Operation While Legal Hold Active
+    let response = blob_client.delete(None).await;
+    // Assert
+    let error = response.unwrap_err().http_status();
+    assert_eq!(StatusCode::Conflict, error.unwrap());
+
+    // Remove Legal Hold
+    blob_client.set_legal_hold(false, None).await?;
+    let response = blob_client.get_properties(None).await?;
+    // Assert
+    let legal_hold = response.legal_hold()?;
+    assert!(!legal_hold.unwrap());
+
+    // Even after removing the legal hold:
+    // Error: Error { context: Full(Custom { kind: HttpResponse { status: Conflict, error_code: Some("ContainerImmutableStorageWithVersioningEnabled") },
+    // error: HttpError { status: Conflict, details: ErrorDetails { code: Some("ContainerImmutableStorageWithVersioningEnabled"), message: None },
+    // body: <?xml version="1.0" encoding="utf-8"?><Error><Code>ContainerImmutableStorageWithVersioningEnabled</Code><Message>
+    // The requested operation is not allowed as the container has a immutable storage with versioning enabled.
+    // Seems like in Python testing, we need the management client in order to delete the container
+    Ok(())
+}
+
+#[recorded::test]
+async fn test_immutability_policy(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    //TODO: Need separate accounts, need a test account with versioning enabled
+    // Recording Setup
+    let recording = ctx.recording();
+    let container_client = get_container_client(recording, false).await?;
+    let blob_client = container_client.blob_client(get_blob_name(recording));
+    container_client.create_container(None).await?;
+    create_test_blob(&blob_client).await?;
+    let test_expiry_time = Some(OffsetDateTime::now_utc() + Duration::minutes(5));
+
+    // Set Immutability Policy
+    let immutability_policy_options = BlobClientSetImmutabilityPolicyOptions {
+        immutability_policy_expiry: test_expiry_time.clone(),
+        immutability_policy_mode: Some(ImmutabilityPolicyMode::Unlocked),
+        ..Default::default()
+    };
+    let response = blob_client
+        .set_immutability_policy(Some(immutability_policy_options))
+        .await?;
+
+    // Assert
+    // ERRORS: Error: Error { context: Full(Custom { kind: DataConversion, error: Error { context: Message { kind: DataConversion, message: "unknown variant of BlobImmutabilityPolicyMode found: \"unlocked\"" } } },
+    // "unable to parse header 'HeaderName(\"x-ms-immutability-policy-mode\"): HeaderValue'
+    // into azure_storage_blob::generated::models::enums::BlobImmutabilityPolicyMode") }
+    // let mode = response.immutability_policy_mode()?;
```

Comment:
This is an interesting one-- we have `ImmutabilityPolicyMode` and `BlobImmutabilityPolicyMode`. The difference is that the `Blob` prefix one contains `Mutable` which is a valid enum for the server to return, but not a valid value for us to set.

However, there seems to be some issue with converting it here?